AppCenter/AppCenterTests/MSEncrypterTests.m

```diff
@@ -17,80 +19,411 @@ @implementation MSEncrypterTests
 - (void)setUp {
   [super setUp];
   self.keychainUtilMock = [MSMockKeychainUtil new];
-  self.keyTag = @"kMSTestEncryptionKeyTag";
 }
 
 - (void)tearDown {
   [self.keychainUtilMock stopMocking];
-  [MSEncrypter deleteKeyWithTag:self.keyTag];
+  [MSMockKeychainUtil clear];
 }
 
-- (void)testEncryption {
+- (void)testEncryptWithCurrentKey {
 
   // If
-  MSEncrypter *encrypter = [[MSEncrypter alloc] initWitKeyTag:self.keyTag];
-  NSString *stringToEncrypt = @"Test string";
+  NSString *clearText = @"clear text";
+  NSString *keyTag = kMSEncryptionKeyTagAlternate;
+  NSString *expectedMetadata = [NSString stringWithFormat:@"%@/AES/CBC/PKCS7/32", keyTag];
+
+  // Save metadata to user defaults.
+  MSMockUserDefaults *mockUserDefaults = [MSMockUserDefaults new];
+  NSDate *expiration = [NSDate dateWithTimeIntervalSinceNow:10000000];
+  NSString *expirationIso = [MSUtility dateToISO8601:expiration];
+  NSString *keyMetadataString = [NSString stringWithFormat:@"%@/%@", keyTag, expirationIso];
+  [mockUserDefaults setObject:keyMetadataString forKey:kMSEncryptionKeyMetadataKey];
+
+  // Save key to the Keychain.
+  NSString *currentKey = [self generateTestEncryptionKey];
+  [MSMockKeychainUtil storeString:currentKey forKey:keyTag];
```

Comment:
You can use `MSKeychainUtil` instead.

Reply:
When should MSMockKeychainUtil be used? I don't think we should actually use a keychain on the test device

Reply:
Unlike `MSMockUserDefaults` `MSMockKeychainUtil` is class mocking `MSKeychainUtil` so that you can just use `MSKeychainUtil` directly. We are not using `MSMockKeychainUtil` directly.

Reply:
But i think this way works too, no?

Reply:
Eventually it would be same. The stubs are actually calling methods in the mock class.

Reply:
Got it, good to know!